src/swdmt/redistricting/RegionTest.java

```diff
@@ -86,4 +88,40 @@ public void squareRegionNumberOfVotersTest() {
                        (new Region(i * i)).numberOfVoters(), is(i * i));
         }
     }
+    @Test(timeout = MAX_TIMEOUT)
+    public void regionWithVotersInsideLocations() {
+        Location location1 = new Location(3, 3);
+        Location location2 = new Location(0, 1);
+        Voter voter1 = new Voter(Party.NONE, location1);
+        Voter voter2 = new Voter(Party.PARTY0, location2);
+
+        Set<Location> locationSet = new HashSet<Location>();
+        locationSet.add(location1);
+        locationSet.add(location2);
+
+        Set<Voter> voterSet = new HashSet<Voter>();
+        voterSet.add(voter1);
+        voterSet.add(voter2);
+
+        Region region = new Region(locationSet, voterSet);
+        assertThat("Region with Voters inside Locations has wrong size ", region.size(), is(2));
+        assertThat("Region with Voters inside Locaton has wrong number of voters ", region.numberOfVoters(), is(2));
+
+    }
+
+    @Test(timeout = MAX_TIMEOUT)
+    public void regionWithVotersOutsideLocations() {
+        Location location3 = new Location(0,3);
+        Voter voter3 = new Voter(Party.NONE, new Location(0,2));
+
+        Set<Location> locationSet = new HashSet<Location>();
+        locationSet.add(location3);
+
+        Set<Voter> voterSet = new HashSet<Voter>();
+        voterSet.add(voter3);
+
+        Region region = new Region(locationSet, voterSet);
+        assertThat("Region with Voters outside of Locations has wrong size ", region.size(), is(1));
+        assertThat("Region with Voters outside of Location has wrong number of voters ", region.numberOfVoters(), is(0));
```

Comment:
Change error message to: "Region with no voters in the location set test failed; given Location set size 1 and Voter set size 1 produced wrong number of voters in the region."